modules/core/include/opencv2/core/persistence.hpp

```diff
@@ -340,6 +340,13 @@ class CV_EXPORTS_W FileStorage
      */
     CV_WRAP FileStorage(const String& filename, int flags, const String& encoding=String());
 
+    /** @brief Copy constructor for FileStorage
+
+    Copy constructor that creates deep copies of passed in cv::FileStorage objects. Based off the full
+    constructor.
```

Comment:
Almost all non-trivial OpenCV classes are designed as a smart pointers (this behavior is aligned with other language bindings):
- assignment operation means increasing object lifetime (through refcounter)
- optional deep-copy is through `.clone()` operations

Deep copy is over-complicated thing for `FileStorage` (especially in case of write mode), so let avoid that completely.